lib/test_track.rb

```diff
@@ -52,6 +54,27 @@ def misconfiguration_notifier_class_name=(notifier_class_name)
       @misconfiguration_notifier_class_name = notifier_class_name
     end
 
+    def load_build_timestamp # rubocop:disable Metrics/MethodLength
+      if Rails.env.test? || Rails.env.development?
+        @build_timestamp = Time.zone.now.iso8601
+      elsif File.exist?(BUILD_TIMESTAMP_FILE_PATH)
```

Comment:
I'm thinking we can maybe combine the error cases so we only have to `raise` once (regardless of whether the file exists)


```ruby
    def load_build_timestamp
      if Rails.env.test? || Rails.env.development?
        @build_timestamp = Time.zone.now.iso8601
      elsif _build_timestamp
        unless BUILD_TIMESTAMP_REGEX.match?(timestamp)
          raise "./testtrack/build_timestamp is not a valid ISO 8601 timestamp, got '#{timestamp}'"
        end

        @build_timestamp = _build_timestamp
      else
        raise '...'
      end
    end

    attr_reader :build_timestamp

  # ...

  def _build_timestamp
    File.read(BUILD_TIMESTAMP_FILE_PATH).chomp.presence if File.exist?(BUILD_TIMESTAMP_FILE_PATH)
  end
```

Main thing there is that by calling `.presence` sooner and sharing the `raise` during the initializer code, we can make `build_timestamp` an `attr_reader` and read straight off the ivar.

Reply:
(I'm thinking this is a safe strategy because the gem itself defines the initializer and owns `load_build_timestamp`, so unless we suspect that somehow folks are gonna skip initialization of the app?)

Reply:
(Alternatively, I sort of wonder if we could rename `load_build_timestamp` to `build_timestamp` -- have it fetch all the things internally, run all the checks, etc -- and then basically just pre-heat it from the initializer.)